File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
answer = 0 

def dfs(n, numbers, target, depth, now):
    global answer
    
    if depth == n-1:
        if now + numbers[depth] == target:
            answer += 1
        elif now - numbers[depth] == target:
            answer += 1
        return 
    dfs(n, numbers, target, depth + 1, now+numbers[depth])
    dfs(n, numbers, target, depth + 1, now-numbers[depth])
    
    return

def solution(numbers, target):
    global answer
    dfs(len(numbers), numbers, target, 0, 0)
    
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    
    function dfs(sumValue, depth){
        if (depth === numbers.length) {
            if (sumValue === target) answer++;
            return;
        };
        dfs(sumValue + numbers[depth], depth+1);
        dfs(sumValue - numbers[depth], depth+1);    
    }
    
    dfs(0, 0)
    return answer;
}